ci/infra/azure/variables.tf

```diff
@@ -173,3 +173,8 @@ variable "peer_virutal_network_id" {
   default     = []
   description = "IDs of a Virtual Network to connect to via a peering connection"
 }
+
+variable "cpi_enable" {
+  default     = false
+  description = "Enable CPI integration with Azure"
+}
```

Comment:
I did not saw `cpi_enable` in AWS. So I don't know what's the purpose for let users determine `cpi_enable=false` because I thought for public cloud, the CPI always on.
Could you please elaborate more.

Reply:
Currently we support deployment with and without cloud provider. If all agree I can remove `cpi_enable` and consolidate the resource creation. WDYT @innobead , @cclhsu .

Reply:
> I did not saw `cpi_enable` in AWS. So I don't know what's the purpose for let users determine `cpi_enable=false` because I thought for public cloud, the CPI always on.
> Could you please elaborate more.

A good question to rethink a bit.

There is no AWS cloud provider integration yet in AWS TFs, so this is why `cpi_enable` not there. I think we are implicitly adopting the way coming from the initial examples of `openstack` and `vsphere`, so this is why `cpi_enable` introduced here.

Let's look around the market, `rancher` provides customers to decide if setting the cloud config of a specific infra for integrating the corresponding cloud provider.  ref: https://rancher.com/docs/rke/latest/en/config-options/cloud-providers/. However, openshift 4 seems not to disclose any details about it, but 3 does the way as Rancher RKE. 

I would suggest let's keep this design to have two options for users like rancher does, and look for any feedback in the future to see if we should stick on always enabling cloud provider integration. Also, this can keep the same design as what we already provide like `openstack` and `vsphere`.

- disable cloud provider, and use on-prem like deployment on cloud
- enable cloud provider, and use cloud provider integration

Reply:
Cool, because I was thinking if _without_ cloud provider enabled in public cloud, how the user enables ingress & load balancer related function :thinking: .
But I'm lack of knowledge for this because my past experience was always enabled for the public cloud.

But as you mentioned we did not have AWS cloud provider integration yet in AWS TFs. But if I remember correctly, after AWS deployment, I could see the external IPs from `kubectl get nodes -o wide`. So I'm not sure if AWS TFs by default is _cpi_enable=true_ or _cpi_enable=false_.

I agree to keep current design but just curious what's the default value `cpi_enable` should be.

Reply:
Yes, you are right. native ingress integration is provided by Azure cloud provider integration, ref: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/docs/services/README.md.

After discussing with @jenting again, some conclusions to follow up later on.
- deployment: skuba provides optionally enabling cloud provider (as what we have no by providing `--cloud-provider`)
- infra: should we have `cpi_enable` in TFs, or we should have agnostic terraforms scripts regardless if having cloud provider integration or not. To have better user experience. 

To sum up the current cloud providers we have right now. 
- `openstack`: `cpi_enable` is used to automatically generate and copy openstack.conf to each node.
- `vsphere`: cpi_enable` is used to manage `vsphere_folder` resource
- `azure`: @c3y1huang is still working on that. 

@c3y1huang let's see if it's possible to remove `cpi_enable`, and it can be working regardless `--cloud-provider` enabled or not. If yes, this always achieves what we discussed.

And, later we definitely need to revisits `vsphere` and `openstack` to make them consistent behavior.